src/include/pgagroal.h

```diff
@@ -371,10 +371,28 @@ struct prometheus_cache
 } __attribute__ ((aligned (64)));
 
 /** @struct
- * Defines the Prometheus metrics
+ * Defines the common Prometheus metrics
  */
 struct prometheus
 {
+   // logging
+   atomic_ulong logging_info;          /**< Logging: INFO */
+   atomic_ulong logging_warn;          /**< Logging: WARN */
+   atomic_ulong logging_error;         /**< Logging: ERROR */
+   atomic_ulong logging_fatal;         /**< Logging: FATAL */
+
+   // internal connections
+   atomic_int client_sockets;          /**< The number of sockets the client used */
+   atomic_int self_sockets;            /**< The number of sockets used by pgagroal itself */
+
+} __attribute__ ((aligned (64)));
+
+/** @struct
+ * Defines the Main Prometheus metrics
+ */
+struct main_prometheus
```

Comment:
I don't get the point in splitting the `prometheus` structure into two parts, it seems to me it is useless: if `metrics > 0` prometheus is active, so there is no advantage in memory allocation. Also, while `vault_configuration` added fields around `configuration`, here you are narrowing the structure only to the fields you need.
While I don't dislike in total this approach, it seems counterintuitive to have `prometheus` not being the main metrics structure as I would at glange expect. I would rather make a `prometheus_logging` structure, a `prometheus_socket` one and have `prometheus` to wrap the above, so that `prometheus_vault` can handle only the former twos.

Reply:
> if metrics > 0 prometheus is active, so there is no advantage in memory allocation.

True, the aim of splitting `prometheus` is to separate the memory space allocation for different programs (`main`/`vault`) as there as several fields which are not utilized by vault but are still allocated extra memory for unused fields.

> Also, while vault_configuration added fields around configuration, here you are narrowing the structure only to the  
> fields you need.

As of now I have added just the required fields and more fields can be added as and when required.

> While I don't dislike in total this approach, it seems counterintuitive to have prometheus not being the main metrics  
> structure as I would at glange expect. I would rather make a prometheus_logging structure, a prometheus_socket 
> one and have prometheus to wrap the above, so that prometheus_vault can handle only the former twos.

Can be done that way also. The thing i kept in my mind was that if in future if the project integrates other such service programs like `vault`, that will need `prometheus` then it can directly inherit the common fields (common for all services/servers) while keep on adding its specific fields in its `prometheus` structure.

Reply:
> Can be done that way also. The thing i kept in my mind was that if in future if the project integrates other such service programs like `vault`, that will need `prometheus` then it can directly inherit the common fields (common for all services/servers) while keep on adding its specific fields in its `prometheus` structure.

This is clear, and is a good approach, but I don't think that `common` is the right name for the structure. It seems, so far, you need only a subset of the structure for the vault, so I'm not sure if this effort in splitting the structure is worth. I suspect you will find the need to add more fields here and there from time to time, so at this stage I would not split the structure. Surely, I will not name it as `common`, since that makes me think at *common* fields, while so far there are only "basic" fields like logging counters.
@jesperpedersen what do you think?

Reply:
`prometheus_base` is an alternative